numba/npyufunc/parfor.py

```diff
@@ -660,6 +660,21 @@ def redarraytype_to_sig(redarraytyp):
     assert isinstance(redarraytyp, types.npytypes.Array)
     return types.npytypes.Array(redarraytyp.dtype, max(1, redarraytyp.ndim - 1), redarraytyp.layout)
 
+def legalize_names_with_typemap(names, typemap):
+    """ We use ir_utils.legalize_names to replace internal IR variables that have things like
```

Comment:
I think I understand this sentence but it was quite hard to parse. Is the intent:
> We use ir_utils.legalize_names to replace internal IR variable names containing illegal characters (e.g. period) with a legal character (underscore) so as to create legal variable names.

?

Reply:
Done.